src/crimson/osd/osd_operations/peering_event.h

```diff
@@ -61,6 +61,7 @@ class PeeringEvent : public OperationT<PeeringEvent> {
 
   virtual void on_pg_absent();
   virtual PeeringEvent::interruptible_future<> complete_rctx(Ref<PG>);
+  virtual seastar::future<> complete_rctx_no_pg() { return seastar::now();}
```

Comment:
I think this is misleading.  The path in question here can't actually have generated any messages in the rctx, I don't think.  If it can, then defaulting to discarding them here would be wrong.  I think we can remove the complete_rctx_no_pg() calls entirely.

Reply:
@rzarzynski ?

Reply:
@athanatos there is ctx messages when remote peering event, for local peer event do nothing, for remote peer event will send message to peer.  There is RemotePeeringEvent::complete_rctx_no_pg which is inherit from this virtual PeeringEvent::complete_rctx_no_pg.